src/java.base/share/classes/java/lang/StackFrameInfo.java

```diff
@@ -28,16 +28,17 @@
 import jdk.internal.access.SharedSecrets;
 import jdk.internal.vm.ContinuationScope;
 
-import java.lang.StackWalker.StackFrame;
 import java.lang.invoke.MethodType;
+import java.lang.reflect.Modifier;
 
-class StackFrameInfo implements StackFrame {
+class StackFrameInfo extends ClassFrameInfo {
     private static final JavaLangInvokeAccess JLIA =
         SharedSecrets.getJavaLangInvokeAccess();
 
     private final boolean retainClassRef;
```

Comment:
Shouldn't `private final boolean retainClassRef` move to ClassFrameInfo? Isn't it strange that CLASS_INFO still implies RETAIN_CLASS_REFERENCE?

Reply:
I was planning to move `retainClassRef` to `ClassFrameInfo` as a bit set in the `flags` field.   Thanks for catching this and reminding me to do that.  Now updated.